### ISSUE TITLE
main keybindings file/ changing keybindings

### ISSUE THREAD

**Issue by d0sensuppe** [NsCDE/NsCDE#101]
Really impressed with your work. No issues so far.

I don't want to ask anything reduntant here, but I couldn't find anything in the docs concerning my issue.
It's just that I have been using ~/.NsCDE/Keybindings.fvwmlocal so far to set my bindings, but now I have some reduntant functions and bindings that I don't want. 
I have been using it more as the main keybindings file as which it propably isn't intendended.

• Third form of FVWM configuration in $NSCDE_USERDIR are **".fvwmlocal" files.** These files will **extend (append)** configuration read from their ".fvwmconf" and/or ".fvwmgen" files. 

Also don't know if this is related:
• Files with extension ".fvwmconf" are overriding defaults from the $NSCDE_DATADIR/fvwm
In other words, **if $NSCDE_USERDIR/Init.fvwmconf exists, it will be read instead of $NSCDE_DATADIR/fvwm/Init.fvwmconf** 
Maybe I need to put it in some subdirectory in ~/.NsCDE ?

tl;dr

Deleting /usr/share/NsCDE/fvwm/Keybindings.nscde or moving it into my ~ folder does nothing.
How can I edit it directly and use as my main keybindings config?

Best regards

**Comment by d0sensuppe** [comment]
Thanks for your reply. 

The keybindings I set in ~fvwmlocal work. 
It's just that deleting $NSCDE_DATADIR/fvwm/Keybindings.nscde seemingly doesn't do anything.
I thought the config files in ~ would replace everything specified in DATADIR. 
Am I missing something here? Sorry I didn't have the time yet to go through the whole documentaton in detail.

Anyways here is my Keybindings.fvwmlocal. I know it's a bit of a mess. 
This gaussianlock is only a simple bashscript creating a blur effect using i3 lock that I've thrown in there instead of xlock:
`# _______________________________________________________________________
# Key/Mouse (window) | button/key name | context | modifiers | action
#		WINDOW
Key 				Q 	   A		S4	ThisWindow (!FvwmButtons) Close
Key				X	   A 		4       ThisWindow (!FvwmButtons) Iconify
Key 				A	   A 		4	TileLeft
Key 				D	   A 		4	TileRight
Key 				W	   A 		4	TileTop
Key 				S	   A 		4	TileBottom
Key 				Q	   A 		4	TileTopLeft
Key 				E	   A 		4	TileTopRight
Key 				Y	   A 		4	TileBottomLeft
Key 				C	   A 		4	TileBottomRIght
Key				V	   A 		4	Move
Key				R	   A		4	Raise
Key 				1	   A		S4	MoveToDesk 0 0
Key				2	   A		S4	MoveToDesk 0 1  
Key				3	   A		S4	MoveToDesk 0 2
Key				4	   A		S4	MoveToDesk 0 3
Key				5	   A		S4	MoveToDesk 0 4
Key				6	   A		S4	MoveToDesk 0 5
Key				7	   A		S4	MoveToDesk 0 6
Key				8	   A		S4	MoveToDesk 0 7
Key 				1	   A		C4	Desk 0 0
Key 				2	   A		C4	Desk 0 1
Key 				3	   A		C4	Desk 0 2
Key 				4	   A		C4	Desk 0 3
Key 				5	   A		C4	Desk 0 4
Key 				6	   A		C4	Desk 0 5
Key 				7	   A		C4	Desk 0 6
Key 				8	   A		C4	Desk 0 7
Key				D	   A		C4	GotoDesk 1 0 3
Key				A	   A		C4	GotoDesk -1 0 3
Key				Y	   A		S4	Stick
Key 				Left       A   	        4   	Direction West (CurrentDesk, AcceptsFocus, Visible) Focus
Key 				Right      A   	        4   	Direction East (CurrentDesk, AcceptsFocus, Visible) Focus
Key 				Up         A   	        4   	Direction North (CurrentDesk, AcceptsFocus, Visible) Focus
Key 				Down       A  	        4   	Direction South (CurrentDesk, AcceptsFocus, Visible) Focus
Key				W	   A		S4 	WindowMax
Key 				Space	   A		4	f_TileWindows
Key 				Tab 	   A 		4 	f_SwitchFocus Next CurrentPage
Key				Tab	   A		C4	WindowList

#		APPLICATIONS
Key 				L	   A 		S4	Exec exec bash ~/.NsCDE/gaussianlock.sh
Key				L	   A		C4	I3Suspend
Key 				D	   A		S4	Exec exec rofi -show run
Key				E	   A		S4	Exec exec nautilus
Key				S	   A		S4	Exec exec xfce4-terminal 
Key				K	   A		S4	Exec exec keepassxc


# 		FUNCTIONS
# _______________________________________________________________________
#	Tiling
DestroyFunc TileLeft
AddToFunc TileLeft
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move +0 +0
+ I ThisWindow (!FvwmButtons, !Maximized, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move +0 +0a
+ I WarpToWindow 5p 5p

DestroyFunc TileRight
AddToFunc TileRight
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move -0 +0
+ I ThisWindow (!FvwmButtons, !Maximized, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move -0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileTop
AddToFunc TileTop
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 100 50 +0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileBottom
AddToFunc TileBottom
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 100 50 +0 -0
+ I WarpToWindow 5p 5p

DestroyFunc TileTopLeft
AddToFunc TileTopLeft
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 +0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileTopRight
AddToFunc TileTopRight
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 -0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileBottomLeft
AddToFunc TileBottomLeft
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 +0 -0
+ I WarpToWindow 5p 5p

DestroyFunc TileBottomRight
AddToFunc TileBottomRight
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 -0 -0
+ I WarpToWindow 5p 5p

# _______________________________________________________________________
#	Maximize Window
DestroyFunc WindowMax
AddToFunc WindowMax
+ I ThisWindow (!Shaded, !Iconic) Maximize 100 -1p
+ I ThisWindow (!Maximized, !Shaded, !Iconic) Maximize 100 -1p
# _______________________________________________________________________
#	Lock and Suspend
DestroyFunc I3Suspend
AddToFunc I3Suspend
+ I Exec exec systemctl suspend
+ I Exec exec bash ~/.NsCDE/gaussianlock.sh
`

**Comment by d0sensuppe** [comment]
Thanks for your reply. 

The keybindings I set in ~fvwmlocal work. 
It's just that deleting $NSCDE_DATADIR/fvwm/Keybindings.nscde seemingly doesn't do anything.
I thought the config files in ~ would replace everything specified in DATADIR. 
Am I missing something here? Sorry I didn't have the time yet to go through the whole documentaton in detail.

Anyways here is my Keybindings.fvwmlocal. I know it's a bit of a mess. 
This gaussianlock is only a simple bashscript creating a blur effect using i3 lock that I've thrown in there instead of xlock:
```
# _______________________________________________________________________
# Key/Mouse (window) | button/key name | context | modifiers | action
#		WINDOW
Key 				Q 	   A		S4	ThisWindow (!FvwmButtons) Close
Key				X	   A 		4       ThisWindow (!FvwmButtons) Iconify
Key 				A	   A 		4	TileLeft
Key 				D	   A 		4	TileRight
Key 				W	   A 		4	TileTop
Key 				S	   A 		4	TileBottom
Key 				Q	   A 		4	TileTopLeft
Key 				E	   A 		4	TileTopRight
Key 				Y	   A 		4	TileBottomLeft
Key 				C	   A 		4	TileBottomRIght
Key				V	   A 		4	Move
Key				R	   A		4	Raise
Key 				1	   A		S4	MoveToDesk 0 0
Key				2	   A		S4	MoveToDesk 0 1  
Key				3	   A		S4	MoveToDesk 0 2
Key				4	   A		S4	MoveToDesk 0 3
Key				5	   A		S4	MoveToDesk 0 4
Key				6	   A		S4	MoveToDesk 0 5
Key				7	   A		S4	MoveToDesk 0 6
Key				8	   A		S4	MoveToDesk 0 7
Key 				1	   A		C4	Desk 0 0
Key 				2	   A		C4	Desk 0 1
Key 				3	   A		C4	Desk 0 2
Key 				4	   A		C4	Desk 0 3
Key 				5	   A		C4	Desk 0 4
Key 				6	   A		C4	Desk 0 5
Key 				7	   A		C4	Desk 0 6
Key 				8	   A		C4	Desk 0 7
Key				D	   A		C4	GotoDesk 1 0 3
Key				A	   A		C4	GotoDesk -1 0 3
Key				Y	   A		S4	Stick
Key 				Left       A   	        4   	Direction West (CurrentDesk, AcceptsFocus, Visible) Focus
Key 				Right      A   	        4   	Direction East (CurrentDesk, AcceptsFocus, Visible) Focus
Key 				Up         A   	        4   	Direction North (CurrentDesk, AcceptsFocus, Visible) Focus
Key 				Down       A  	        4   	Direction South (CurrentDesk, AcceptsFocus, Visible) Focus
Key				W	   A		S4 	WindowMax
Key 				Space	   A		4	f_TileWindows
Key 				Tab 	   A 		4 	f_SwitchFocus Next CurrentPage
Key				Tab	   A		C4	WindowList

#		APPLICATIONS
Key 				L	   A 		S4	Exec exec bash ~/.NsCDE/gaussianlock.sh
Key				L	   A		C4	I3Suspend
Key 				D	   A		S4	Exec exec rofi -show run
Key				E	   A		S4	Exec exec nautilus
Key				S	   A		S4	Exec exec xfce4-terminal 
Key				K	   A		S4	Exec exec keepassxc


# 		FUNCTIONS
# _______________________________________________________________________
#	Tiling
DestroyFunc TileLeft
AddToFunc TileLeft
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move +0 +0
+ I ThisWindow (!FvwmButtons, !Maximized, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move +0 +0a
+ I WarpToWindow 5p 5p

DestroyFunc TileRight
AddToFunc TileRight
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move -0 +0
+ I ThisWindow (!FvwmButtons, !Maximized, !Shaded, !Iconic) Maximize 50 100
+ I WarpToWindow 5p 5p
+ I ThisWindow (!FvwmButtons, Maximized, !Shaded, !Iconic) Move -0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileTop
AddToFunc TileTop
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 100 50 +0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileBottom
AddToFunc TileBottom
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 100 50 +0 -0
+ I WarpToWindow 5p 5p

DestroyFunc TileTopLeft
AddToFunc TileTopLeft
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 +0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileTopRight
AddToFunc TileTopRight
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 -0 +0
+ I WarpToWindow 5p 5p

DestroyFunc TileBottomLeft
AddToFunc TileBottomLeft
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 +0 -0
+ I WarpToWindow 5p 5p

DestroyFunc TileBottomRight
AddToFunc TileBottomRight
+ I ThisWindow (!FvwmButtons, !Shaded, !Iconic) ResizeMoveMaximize 50 50 -0 -0
+ I WarpToWindow 5p 5p

# _______________________________________________________________________
#	Maximize Window
DestroyFunc WindowMax
AddToFunc WindowMax
+ I ThisWindow (!Shaded, !Iconic) Maximize 100 -1p
+ I ThisWindow (!Maximized, !Shaded, !Iconic) Maximize 100 -1p
# _______________________________________________________________________
#	Lock and Suspend
DestroyFunc I3Suspend
AddToFunc I3Suspend
+ I Exec exec systemctl suspend
+ I Exec exec bash ~/.NsCDE/gaussianlock.sh


```

**Comment by d0sensuppe** [comment]
Sorry for being lazy and wasting your time.
I am using ~/.nsCDE/Keybindings.fvwmconf instead of fvwmlocal.
> 
> Keybindings can be partially or completely overriden by the user’s writing
> $FVWM_USERDIR/Keybindings.fvwmconf (override) or
> $FVWM_USERDIR/Keybindings.fvwmlocal (extend, paritally override). Key bindings which are
> described with their Mod+Key items on menus can be extended or overwritten in
> $FVWM_USERDIR/Keymenu.actions
